processor/src/main/java/com/blarley/rxretrobusseed/annotationprocessor/processor/RxRetroBusAnnotationProcessor.java

```diff
@@ -71,7 +71,7 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
             for (Element subElement : roundEnv.getElementsAnnotatedWith(Publish.class)) {
 
                 // ExecutableElements represent methods (among other things) - TODO: Figure out how this can break
-                if (subElement instanceof ExecutableElement) {
+                if (subElement instanceof ExecutableElement && subElement.getEnclosingElement() == element) {
```

Comment:
Does element have a getElementsAnnotatedWith method as well, or something of the sort?